### ISSUE TITLE
Have "at" default to the next applicable date

### ISSUE THREAD

**Issue by DevL** [magplus/good_times#22]
Instead of applying a time to a specified datetime, `at` could default to the next applicable date.

Defaulting to `now` only brings us part of the way there as the time can already have been passed this day.

A few examples might clarify my chain of thoughts:

```elixir
now              # => {{2015, 3, 10}, {7, 43, 50}}
at {10, 0, 0}  # => {{2015, 3, 10}, {10, 0, 0}}
at {6, 30, 0}  # => {{2015, 3, 11}, {6, 30, 0}}
```

**Comment by martinsvalin** [comment]
I don't think the name `at` really sets up this expectation. For me, it's "here is a date and time, give me that, but where the time is X". So my expectation is for `{{2015, 2, 27}, {18, 30, 45}} |> at {12, 0, 0}` to give me `{{2015, 2, 27}, {12, 0, 0}}` wether I call it in the morning or evening.

What use case lead you in this direction?

**Comment by martinsvalin** [comment]
Closed by #21

**Comment by DevL** [comment]
Convenience. And the idea of "let's schedule a meeting at 12:00" which would be the next applicable "12:00". But since it's not self-evident what it would do I agree with closing this.
